src/Acuminator/Acuminator.Analyzers/Refactorings/EventHandlerSignature/EventHandlerSignatureRefactoring.cs

```diff
@@ -0,0 +1,160 @@
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Utilities.Roslyn.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Acuminator.Analyzers.Refactorings.EventHandlerSignature
+{
+	[ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+	public class EventHandlerSignatureRefactoring : CodeRefactoringProvider
+	{
+		private const string ArgsParameterName = "e";
+		private const string EventHandlerMethodName = "_";
+
+		public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+			var pxContext = new PXContext(semanticModel.Compilation);
+
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+			var methodNode = root?.FindNode(context.Span)?.GetDeclaringMethodNode();
+
+			if (methodNode != null)
+			{
+				IMethodSymbol methodSymbol = semanticModel.GetDeclaredSymbol(methodNode, context.CancellationToken);
+
+				if (IsSuitableForRefactoring(methodSymbol, pxContext))
+				{
+					var eventHandlerInfo = methodSymbol.GetEventHandlerInfo(pxContext);
+
+					if (eventHandlerInfo.eventSignatureType == EventHandlerSignatureType.Default
+						&& pxContext.Events.EventHandlerSignatureTypeMap.TryGetValue(
+							(eventHandlerInfo.eventType, EventHandlerSignatureType.Generic), out var genericArgsSymbol)
+						&& methodSymbol.Name.EndsWith("_" + eventHandlerInfo.eventType, StringComparison.Ordinal))
+					{
+						(string dacName, string fieldName) = ParseMethodName(methodSymbol.Name, eventHandlerInfo.eventType);
+
+						string title = nameof (Resources.EventHandlerSignatureCodeActionTitle).GetLocalized().ToString();
+						context.RegisterRefactoring(CodeAction.Create(title,
+							ct => ChangeSignatureAsync(context, root, semanticModel, methodNode, methodSymbol,
+								eventHandlerInfo.eventType, genericArgsSymbol, dacName, fieldName, ct), title));
+					}
+				}
+			}
+		}
+
+		private bool IsSuitableForRefactoring(IMethodSymbol symbol, PXContext pxContext)
+		{
+			return symbol?.ContainingType?.OriginalDefinition != null
+			       && symbol.ContainingType.OriginalDefinition.IsPXGraphOrExtension(pxContext)
+			       && symbol.Parameters.Length <= 2
+			       && !symbol.IsOverride
+				   && !symbol.GetAttributes().Any(
+					   attr => attr.AttributeClass != null && attr.AttributeClass.Equals(pxContext.AttributeTypes.PXOverrideAttribute));
+		}
+
+		private Task<Document> ChangeSignatureAsync(CodeRefactoringContext context, 
```

Comment:
@dropsonic you don't need to pass the whole `CodeRefactoringContext` to this method, you just need a Document.
Also there are a lot of parameters in this method. Can't their number be reduced like obtaining semantic model from Document once again? (we already removed caching of semantic model in other places).

Reply:
@SENya1990 it is possible, but it might affect performance